app/src/main/java/org/p2p/wallet/pnl/storage/PnlStorage.kt

```diff
@@ -0,0 +1,55 @@
+package org.p2p.wallet.pnl.storage
+
+import android.content.SharedPreferences
+import com.google.gson.Gson
+import java.time.ZonedDateTime
+import org.p2p.core.crypto.Base58String
+import org.p2p.core.network.gson.gsonGenericMapType
+import org.p2p.core.utils.fromJsonReified
+import org.p2p.wallet.pnl.models.PnlTokenData
+
+class PnlStorage(
+    private val sharedPreferences: SharedPreferences,
+    private val gson: Gson,
+) : PnlStorageContract {
+    private companion object {
+        const val KEY_TOTAL = "TOTAL"
+        const val KEY_TOKENS = "TOKENS"
+        const val KEY_LAST_UPDATE_TIME = "LAST_UPDATE_TIME"
+    }
+
+    override var total: PnlTokenData
+        get() {
+            return sharedPreferences
```

Comment:
can we extract gets and sets into methods so we make it consistent with other storages?